pic32/cores/pic32/WInterrupts.c

```diff
@@ -263,12 +265,22 @@ void __attribute__((nomips16,at_vector(_EXTERNAL_0_VECTOR),interrupt(IPL4SRS)))
 void __attribute__((interrupt(),nomips16)) ExtInt0Handler(void)
 #endif
 {
-
+	VEC_ENT_CODE(EXT_INT0,INT0IF,PIN_INT0);
```

Comment:
Personally I would run in edge mode, and trigger on rising edge. When triggered you service one interrupt then clear the associated IF flag followed by the associated interrupt flag in the MAX3421E. If there is still a pending interrupt a new INT pulse will be generated so your ISR can run again - and again until all the interrupts have been serviced.